### PR TITLE
Add netstandard1.6 support

### PR DESCRIPTION
**Pull request by Sphiecoh**

### CODE REVIEW

**Comment by khellang** [comment]
I think we can use [`ServiceStack.Text.Core`](https://www.nuget.org/packages/ServiceStack.Text.Core) for the `netstandard` target :smile:

File: src/Nancy.Serialization.ServiceStack/Helpers.cs [hunk 1]
```diff
@@ -23,10 +23,10 @@ public static bool IsJsonType(string contentType)
 
             var contentMimeType = contentType.Split(';')[0];
 
-            return contentMimeType.Equals("application/json", StringComparison.InvariantCultureIgnoreCase) ||
-                   contentMimeType.Equals("text/json", StringComparison.InvariantCultureIgnoreCase) ||
-                  (contentMimeType.StartsWith("application/vnd", StringComparison.InvariantCultureIgnoreCase) &&
-                   contentMimeType.EndsWith("+json", StringComparison.InvariantCultureIgnoreCase));
+            return contentMimeType.Equals("application/json", StringComparison.OrdinalIgnoreCase) ||
+                   contentMimeType.Equals("text/json", StringComparison.OrdinalIgnoreCase) ||
+                  (contentMimeType.StartsWith("application/vnd", StringComparison.OrdinalIgnoreCase) &&
```
Comment (khellang) [inline src/Nancy.Serialization.ServiceStack/Helpers.cs]: While you're at it, could you remove the `contentMimeType.StartsWith("application/vnd", StringComparison.OrdinalIgnoreCase)` check? I don't think it's necessary.

File: src/Nancy.Serialization.ServiceStack/project.json [hunk 2]
```diff
@@ -18,9 +18,14 @@
     "Nancy": {
       "target": "project"
     },
-    "ServiceStack.Text": "3.9.71"
+    "ServiceStack.Text": "4.5.4"
```
Comment (khellang) [inline src/Nancy.Serialization.ServiceStack/project.json]: I think we can use `ServiceStack.Text.Core` for the netstandard target and get rid of the imports :smile:

**Comment by thecodejunkie** [comment]
Just a heads up. When we expand a project to support `netstandard` then we need to make sure `build.cake` script it properly setup to handle both targets

File: src/Nancy.Serialization.ServiceStack/Helpers.cs (hunk 1 above)
Comment (khellang) [inline src/Nancy.Serialization.ServiceStack/Helpers.cs]: Ideally, we'd like this check to be the same as the default serializer; https://github.com/NancyFx/Nancy/blob/master/src/Nancy/Responses/DefaultJsonSerializer.cs#L100-L102

**Comment by Sphiecoh** [comment]
I think we good to GO gents.

**Comment by jchannon** [comment]
I assume we are still planning on supporting the MSBuild folders. If so that will need doing however it seems SStack has different versions per target so not sure how we'll manage that unless we just make packages.config/csproj reference the full fwk version eg/ 4.5.4

File: src/Nancy.Serialization.ServiceStack/Helpers.cs [hunk 3]
```diff
@@ -23,10 +23,9 @@ public static bool IsJsonType(string contentType)
 
             var contentMimeType = contentType.Split(';')[0];
 
-            return contentMimeType.Equals("application/json", StringComparison.InvariantCultureIgnoreCase) ||
```
Comment (khellang) [inline src/Nancy.Serialization.ServiceStack/Helpers.cs]: Can't we just remove this class now?

**Comment by jchannon** [comment]
👀

**Comment by jchannon** [comment]
Relooking at this. Could the SSTack Core dependency be used for the net452 target too? It targets netstandard1.1 so thats compatible..I think?

File: src/Nancy.Serialization.ServiceStack/ServiceStackBodyDeserializer.cs [hunk 4]
```diff
@@ -19,7 +19,7 @@ public class ServiceStackBodyDeserializer : IBodyDeserializer
         /// <returns>True if supported, false otherwise</returns>
         public bool CanDeserialize(MediaRange mediaRange, BindingContext context)
         {
-            return Helpers.IsJsonType(mediaRange);
+            return Json.Json.IsJsonContentType(mediaRange);
```
Comment (jchannon) [inline src/Nancy.Serialization.ServiceStack/ServiceStackBodyDeserializer.cs]: This doesn't compile

**Comment by jchannon** [comment]
Ignore that, it doesn't work

**Comment by thecodejunkie** [comment]
Can you resend this once #38 has gone in? Should be in a day or two at latest? Thanks! 👍

**Pull request closed without merging**